wdl-lint/src/rules/shellcheck.rs

```diff
@@ -375,6 +378,93 @@ impl<'a> CommandContext<'a> {
     }
 }
 
+/// Detect embedded quotes surrounding an expression in a string.
+/// For a given expression, it checks through all descendants to see if there
+/// are any name references (variables) that are surrounded by escaped quotes.
+/// In WDL, the parent expressions is an addition (concatenation) operations.
+/// So the escaped quotes are not in a single string literal.
+fn is_quoted(expr: &Expr) -> bool {
+    let mut opened = false;
+    let mut name = false;
+    for c in expr.descendants::<Expr>() {
```

Comment:
This handles expressions like:
`~{if read_group != "" then "-r '" + read_group + "'" else ""}`